app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -16,7 +17,11 @@ def create
       result = OtpVerificationForm.new(current_user, sanitized_otp_code).submit
       post_analytics(result)
       if result.success?
-        handle_valid_otp
+        next_url = nil
+        if UserSessionContext.confirmation_context?(context)
+          next_url = user_next_authentication_setup_path(nil)
```

Comment:
There's already a default param so no need to pass in `nil`
```suggestion
          next_url = user_next_authentication_setup_path
```